adam_core/coordinates/keplerian.py

```diff
@@ -31,263 +37,72 @@
         """
-        return self.sigmas[:, 0]
-
-    @sigma_a.setter
-    def sigma_a(self, value):
-        self._covariances[:, 0, 0] = value**2
-        self._covariances[:, 0, 0].mask = False
-
-    @sigma_a.deleter
-    def sigma_a(self):
-        self._covariances[:, 0, 0] = np.nan
-        self._covariances[:, 0, 0].mask = True
+        return self.covariances.sigmas[:, 4]
 
     @property
-    def sigma_e(self):
+    def sigma_M(self) -> np.ndarray:
         """
-        1-sigma uncertainty in eccentricity
+        1-sigma uncertainty in mean anomaly.
         """
-        return self.sigmas[:, 1]
-
-    @sigma_e.setter
-    def sigma_e(self, value):
-        self._covariances[:, 1, 1] = value**2
-        self._covariances[:, 1, 1].mask = False
-
-    @sigma_e.deleter
-    def sigma_e(self):
-        self._covariances[:, 1, 1] = np.nan
-        self._covariances[:, 1, 1].mask = True
+        return self.covariances.sigmas[:, 5]
 
     @property
-    def sigma_i(self):
+    def q(self) -> np.ndarray:
         """
-        1-sigma uncertainty in inclination
+        Periapsis distance.
         """
-        return self.sigmas[:, 2]
-
-    @sigma_i.setter
-    def sigma_i(self, value):
-        self._covariances[:, 2, 2] = value**2
-        self._covariances[:, 2, 2].mask = False
-
-    @sigma_i.deleter
-    def sigma_i(self):
-        self._covariances[:, 2, 2] = np.nan
-        self._covariances[:, 2, 2].mask = True
-
-    @property
-    def sigma_raan(self):
-        """
-        1-sigma uncertainty in right ascension of the ascending node
-        """
-        return self.sigmas[:, 3]
-
-    @sigma_raan.setter
-    def sigma_raan(self, value):
-        self._covariances[:, 3, 3] = value**2
-        self._covariances[:, 3, 3].mask = False
-
-    @sigma_raan.deleter
-    def sigma_raan(self):
-        self._covariances[:, 3, 3] = np.nan
-        self._covariances[:, 3, 3].mask = True
-
-    @property
-    def sigma_ap(self):
-        """
-        1-sigma uncertainty in argument of periapsis
-        """
-        return self.sigmas[:, 4]
-
-    @sigma_ap.setter
-    def sigma_ap(self, value):
-        self._covariances[:, 4, 4] = value**2
-        self._covariances[:, 4, 4].mask = False
-
-    @sigma_ap.deleter
-    def sigma_ap(self):
-        self._covariances[:, 4, 4] = np.nan
-        self._covariances[:, 4, 4].mask = True
-
-    @property
-    def sigma_M(self):
-        """
-        1-sigma uncertainty in mean anomaly
-        """
-        return self.sigmas[:, 5]
-
-    @sigma_M.setter
-    def sigma_M(self, value):
-        self._covariances[:, 5, 5] = value**2
-        self._covariances[:, 5, 5].mask = False
-
-    @sigma_M.deleter
-    def sigma_M(self):
-        self._covariances[:, 5, 5] = np.nan
-        self._covariances[:, 5, 5].mask = True
-
-    @property
-    def q(self):
-        """
-        Periapsis distance
-        """
-        return self.a * (1 - self.e)
+        return self.a.to_numpy() * (1 - self.e.to_numpy())
```

Comment:
@spenczar Here is an example I think where it could be nice for the getters in `quivr` to return numpy arrays, or have some method to do that. This property will return the periapsis distance as an array, but if I want to access semi-major axis it's returned as a pyarrow object.